File: esykart-product/src/main/java/com/webientsoft/esykart/product/service/CategoryService.java
package com.webientsoft.esykart.product.service;

import com.webientsoft.esykart.common.model.Status;
import com.webientsoft.esykart.common.model.common.FilterModel;
import com.webientsoft.esykart.common.model.common.PaginatedDataModel;
import com.webientsoft.esykart.common.model.product.CategoryModel;

public interface CategoryService {
    
    void save(CategoryModel model);
    void update( int id,CategoryModel model);
    CategoryModel find(Integer id);
    PaginatedDataModel search(FilterModel model);
    void changeStatus(Integer id, Status status);

}

File: esykart-store/esykart-store-merchant/src/main/resources/public/app/components/category/js/categoryService.js
angular.module('altairApp').constant('categoryServiceUrl', '../products/categories')
.service('CategoryService',
['$http','categoryServiceUrl',
function($http, categoryServiceUrl) {
	var service = {};
	service.save = function (data){
    	return $http.post(categoryServiceUrl,data);
    };
    service.update = function(id,data){
		return $http.put(categoryServiceUrl+"/" +id ,data);
    };
    
    service.find = function(id,projection){
		return $http.get(categoryServiceUrl +"/" +id+"?projection="+projection);
    };
    service.findAll = function(page, size,sort,projection){
    	var finalUrl = categoryServiceUrl+"?projection="+projection;
    		finalUrl += "&page="+page;
    		finalUrl += "&size="+size;
     	if(sort!= null && sort.length > 0){
    		$(sort).each(function(){
    			finalUrl += "&sort=";
    			finalUrl += this;
    		});
    	}
		return $http.get(finalUrl);
    };

    service.search = function(filter,projection){
		return $http.post(categoryServiceUrl +"/search?projection="+projection,filter);
    };
    
    service.changeStatus = function(id,status){
    	return $http.patch(categoryServiceUrl +"/" + id + "/changeStatus?status="+status);
    }
    return service;
} 
]);
